src/embed/connection.rs

```diff
@@ -110,7 +115,7 @@ pub async fn process(store: dag::Store, rx: Receiver<Request>, client_id: String
     let mut recv = true;
 
     futures.push(Box::pin(connection_future(
-        &rx,
+        &receiver,
```

Comment:
You and your long names :).

Reply:
I found the var `tx` too confusing given that we use that all over for transaction.

Reply:
+1.